client/device/cloudfmc/create.go

```diff
@@ -0,0 +1,119 @@
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/goutil"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/retry"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/url"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/model/application/applicationstatus"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/model/devicetype"
+	"time"
+)
+
+type CreateInput struct {
+}
+
+func NewCreateInput() CreateInput {
+	return CreateInput{}
+}
+
+type createApplicationBody struct {
+	ApplicationType    string             `json:"applicationType"`
+	ApplicationStatus  string             `json:"applicationStatus"`
+	ApplicationContent applicationContent `json:"applicationContent"`
+}
+
+type applicationContent struct {
+	Type string `json:"@type"`
+}
+
+type CreateOutput = device.CreateOutput
+
+func Create(ctx context.Context, client http.Client, createInp CreateInput) (*CreateOutput, error) {
+
+	client.Logger.Println("creating cloud FMC")
+
+	// 1. POST /aegis/rest/v1/services/targets/applications
+	createApplicationUrl := url.CreateApplication(client.BaseUrl())
+	createApplicationReq := client.NewPost(
+		ctx,
+		createApplicationUrl,
+		createApplicationBody{
+			ApplicationType:   "FMCE",
+			ApplicationStatus: "REQUESTED",
+			ApplicationContent: applicationContent{
+				Type: "FmceApplicationContent",
+			},
+		},
+	)
+	var createApplicationOutp device.CreateOutput
+	err := createApplicationReq.Send(&createApplicationOutp)
+	if err != nil {
+		return nil, err
+	}
+
+	client.Logger.Println("creating cloud FMC device")
+
+	// 2. POST /aegis/rest/v1/services/targets/devices
+	createOutp, err := device.Create(ctx, client, device.NewCreateInputBuilder().
+		Name("FMC").
+		DeviceType(devicetype.CloudFmc).
+		Model(false).
+		ConnectorType("CDG").
+		IgnoreCertificate(goutil.NewBoolPointer(true)).
+		EnableOobDetection(goutil.NewBoolPointer(false)).
+		Build(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	client.Logger.Println("waiting for fmce state machine to be done")
+
+	err = retry.Do(untilApplicationActive(ctx, client),
+		retry.NewOptionsBuilder().
+			Retries(-1).
+			Timeout(30*time.Minute). // usually takes about 15-20 minutes
+			Delay(3*time.Second).
+			EarlyExitOnError(true).
+			Logger(client.Logger).
+			Build(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	client.Logger.Println("cloud FMC application successfully created")
+
+	return createOutp, nil
+}
+
+func untilApplicationActive(ctx context.Context, client http.Client) retry.Func {
+	var unreachable bool
+	var initialUnreachableTime time.Time
+	return func() (bool, error) {
+		fmc, err := application.Read(ctx, client, application.ReadInput{})
+		if err != nil && !errors.Is(err, application.NotFoundError) {
+			return false, err
+		}
+		if fmc.ApplicationStatus == applicationstatus.Unreachable {
```

Comment:
Clever way to work around this until we fix the updateFeatureFlags. Please add a comment in LH-71821 to also update the provider to remove this hack once it's fixed.

Reply:
thanks! Added comments 👍